File: hanoi_tower/hanoi_tower.py
import time

count = 0

def hanoi_tower(n, fr, tmp, to):

    global count  # Make "count" a global variable

    if(n==1):
        count += 1 
        return count
    
    else:
        hanoi_tower(n-1, fr, to, tmp)
        count += 1
        hanoi_tower(n-1, tmp, fr, to)

start = time.time()
hanoi_tower(25, 'A', 'B', 'C')
end = time.time()

time = (end - start)

print("count = {0}, time = {1}" .format(count, time))

File: hanoi_tower/hanoi_tower.c
#include <stdio.h>
#include <time.h>

int count = 0;

void hanoi_tower(int n, char from, char tmp, char to){
    
    if(n==1) count ++;
    //printf("Move disc 1 from %c to %c\n", from, to);

    else{
        hanoi_tower(n-1, from, to, tmp);
        count ++;
        //printf("Move disc %d from %c to %c\n", n, from, to);
        hanoi_tower(n-1, tmp, from, to);
    }
}

int main(){

    clock_t start, end;

    start = clock();
    hanoi_tower(25, 'A', 'B', 'C');
    end = clock();

    float dif = (float)(end - start)/CLOCKS_PER_SEC;

    printf("count = %d, time = %f", count, dif);
     
    return 0;
}
